packages/core/src/lib/passkey/options.ts

```diff
@@ -0,0 +1,124 @@
+import type { Authenticator, InternalOptions } from "../../types"
+import {
+  generateAuthenticationOptions,
+  generateRegistrationOptions,
+} from "@simplewebauthn/server"
+import type {
+  PublicKeyCredentialCreationOptionsJSON,
+  PublicKeyCredentialRequestOptionsJSON,
+} from "@simplewebauthn/server/script/deps"
+import {
+  type Adapter,
+  type AdapterUser,
+  AdapterAccount,
+} from "../../adapters"
+import { randomString } from "../web"
+import { MissingAdapter } from "../../errors"
+import type { PasskeyProviderType } from "../../providers/passkey"
+
+async function getUserAndAuthenticators(
+  options: InternalOptions<PasskeyProviderType>,
+  email?: string
+): Promise<[Authenticator[] | undefined, AdapterUser | null, AdapterAccount | null]> {
+  const { adapter, provider } = options
+
+  // Validate that the adapter is defined and implements the required methods
+  if (!adapter)
+    throw new MissingAdapter(
+      "Passkey provider requires an adapter."
+    )
+
+  // Get the full user from the email
+  const user = email ? await adapter.getUserByEmail(email) : null
+
+  // Find the user's account associated with the provider
+  const accounts = user ? (await adapter.listLinkedAccounts(user.id)) ?? [] : []
+  const account = accounts.find((a) => a.provider === provider.id) ?? null
+
+  // Find the account's authenticators
+  const authenticators = account
+    ? (await adapter.listAuthenticatorsByAccountId(
+      account.providerAccountId
+    )) ?? undefined
+    : undefined
+
+  return [authenticators, user, account]
+}
+
+/**
+ * Generate passkey authentication options.
+ * If a user is provided, their credentials will be used to generate the options.
+ * Otherwise, allow any credentials.
+ *
+ * @param options
+ * @param email Optional user email to use to generate the options.
+ * @returns The options prepared for the client.
+ */
+export async function authenticationOptions(
+  options: InternalOptions<PasskeyProviderType>,
+  email?: string
+): Promise<PublicKeyCredentialRequestOptionsJSON> {
+  const { provider } = options
+
+  // Get the user's authenticators
+  const [authenticators] = await getUserAndAuthenticators(options, email)
+
+  // Generate authentication options
+  const authOptions = await generateAuthenticationOptions({
+    rpID: provider.relayingParty.id,
+    timeout: provider.timeout,
+    allowCredentials: authenticators?.map((a) => ({
+      id: a.credentialID,
+      type: "public-key",
+      transports: a.transports,
+    })),
+    userVerification: "preferred",
```

Comment:
```suggestion
    userVerification: "preferred",
    ...provider.authenticateOptions
```
do you think we should allow the developer to override the authentication/register options?